src/read_fast5.c

```diff
@@ -258,11 +259,18 @@ herr_t fast5_attribute_itr (hid_t loc_id, const char *name, const H5A_info_t  *i
     }
     else if(strcmp("end_reason",name)==0 && H5Tclass==H5T_ENUM){
         flag_new_group_or_new_attribute_read_group = 0;
-//        operator_data->slow5_record->end_reason = value.attr_uint8_t;
-        std::string key = "ns_" + std::string(name); //not stored
-        char warn_message[300];
-        sprintf(warn_message,"Not stored: Attribute %s/%s is not stored yet until we confirm from ONT about its datatype", operator_data->group_name,name);
-        search_and_warn(operator_data,key,warn_message);
+        if(*(operator_data->flag_lossy)==0){
+            if(*(operator_data->flag_header_is_written)==0){
+                const char *enum_labels[] = {"unknown","partial","mux_change","unblock_mux_change","signal_positive","signal_negative"};
```

Comment:
@sashajenner can you please check if this implementation is okay? This is where I add end_reason to aux_map.

Reply:
This is fine. Just a bit hard coded that's all (is it possible to set enum_labels dynamically based on the fast5 file?).

Reply:
will this catch if nanopore changes their enum keys?

Reply:
no lol line 264 is hardcoded